esphome/components/as7341/as7341.cpp

```diff
@@ -0,0 +1,271 @@
+#include "as7341.h"
+#include "esphome/core/log.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace as7341 {
+
+static const char *const TAG = "as7341";
+
+void AS7341Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up AS7341...");
+  LOG_I2C_DEVICE(this);
+
+  // Verify device ID
+  uint8_t id;
+  this->read_byte(AS7341_ID, &id);
+  ESP_LOGCONFIG(TAG, "  Read ID: 0x%X", id);
+  if ((id & 0xFC) != (AS7341_CHIP_ID << 2)) {
+    this->mark_failed();
+    return;
+  }
+
+  // Power on (enter IDLE state)
+  if (!this->enable_power(true)) {
+    ESP_LOGE(TAG, "  Power on failed!");
+    this->mark_failed();
+    return;
+  }
+
+  // Set configuration
+  this->write_byte(AS7341_CONFIG, 0x00);
+  this->setup_atime(this->atime_);
+  this->setup_astep(this->astep_);
+  this->setup_gain(this->gain_);
+}
+
+void AS7341Component::dump_config() {
+  ESP_LOGCONFIG(TAG, "AS7341:");
+  LOG_I2C_DEVICE(this);
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "Communication with AS7341 failed!");
+  }
+  LOG_UPDATE_INTERVAL(this);
+  ESP_LOGCONFIG(TAG, "  Gain: %u", get_gain());
+  ESP_LOGCONFIG(TAG, "  ATIME: %u", get_atime());
+  ESP_LOGCONFIG(TAG, "  ASTEP: %u", get_astep());
+
+  LOG_SENSOR("  ", "F1", this->f1_);
+  LOG_SENSOR("  ", "F2", this->f2_);
+  LOG_SENSOR("  ", "F3", this->f3_);
+  LOG_SENSOR("  ", "F4", this->f4_);
+  LOG_SENSOR("  ", "F5", this->f5_);
+  LOG_SENSOR("  ", "F6", this->f6_);
+  LOG_SENSOR("  ", "F7", this->f7_);
+  LOG_SENSOR("  ", "F8", this->f8_);
+  LOG_SENSOR("  ", "Clear", this->clear_);
+  LOG_SENSOR("  ", "NIR", this->nir_);
+}
+
+float AS7341Component::get_setup_priority() const { return setup_priority::DATA; }
+
+void AS7341Component::update() {
+  read_channels(this->channel_readings_);
```

Comment:
Consistency is key =)
```suggestion
  this->read_channels(this->channel_readings_);
```